File: PoorPigs.cpp
#include<iostream>
#include<vector>
#include<math.h>

using namespace std;

class Solution {
public:
    int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        
        return ceil(log(buckets) / log(minutesToTest / minutesToDie + 1)); 
    }
};

int main() {
	
	int buckets, minutesToDie, minutesToTest;
	cin>>buckets>>minutesToDie>>minutesToTest;
	
	Solution *solution = new Solution();
	cout<<solution->poorPigs(buckets, minutesToDie, minutesToTest);
	
	return 0;
	
}

File: PoorPigs.py
class Solution:
    def poorPigs(self, buckets: int, minutesToDie: int, minutesToTest: int) -> int:
        pigs = 0
        while (minutesToTest // minutesToDie + 1) ** pigs < buckets:
            pigs += 1
        return pigs


if __name__ == '__main__':
    buckets = int(input())
    minutesToDie = int(input())
    minutesToTest = int(input())
    solution = Solution()
    print(solution.poorPigs(buckets, minutesToDie, minutesToTest))
